src/formElementPlugins/dropdownChoice/src/ApplicationView.tsx

```diff
@@ -18,9 +18,16 @@ const ApplicationView: React.FC<ApplicationViewProps> = ({
 
   useEffect(() => {
     onUpdate(value)
-    if (!value && defaultOption !== undefined)
-      onSave({ text: value, optionIndex: options.indexOf(value) })
-    setValue(options[getDefaultIndex(defaultOption, options)])
```

Comment:
there is a slight difference in logic, previously setValue would fire even when if condition wasn't met (haven't tested thought)

Reply:
Yeah, I think the previous logic was actually wrong. It was saving "value", instead of the default option. And value will be NULL to start with. Have replaced it so that it saves the default option rather than value. Condition has not changed (i.e. there's no current value, and there IS a default option specified)

Reply:
ok, so for this fix, was it necessary to only `setValue` if the condition is met ?

Reply:
Yeah, cos if there's already a value, we don't need to set it. (And if there isn't we set one from from the default, but if there's no default, then there's nothing to set)